### PR TITLE
Change DeployAcceptor to TransactionAcceptor

### LINKED ISSUE #4158: Provide TransactionAcceptor
Create a TransactionAcceptor component similar to DeployAcceptor which handles TransactionV1s but which is not wired up to the reactor yet

### PR DESCRIPTION
**Pull request by Fraser999**
This PR updates the `DeployAcceptor` to `TransactionAcceptor` to allow acceptance of `TransactionV1`s and `Deploy`s.

The acceptance criteria for deploys should be unchanged by this PR, and the flow of V1 transactions through the acceptor is almost identical to that of deploys.  The test suite for the acceptor is also effectively unchanged in terms of exercising deploys, and only includes additional tests covering acceptance of V1 transactions.

Closes #4158.

### CODE REVIEW

**Review by EdHastingsCasperAssociation — APPROVED** [review]

File: node/src/components/fetcher/fetcher_impls/deploy_fetcher.rs [hunk 1]
```diff
@@ -56,7 +59,18 @@ impl ItemFetcher<Deploy> for Fetcher<Deploy> {
         effect_builder: EffectBuilder<REv>,
         id: DeployId,
     ) -> Option<Deploy> {
-        effect_builder.get_stored_deploy(id).await
+        effect_builder
+            .get_stored_transaction(TransactionId::from(id))
+            .await
+            .map(|txn| match txn {
```
Comment (rafal-ch) [inline node/src/components/fetcher/fetcher_impls/deploy_fetcher.rs:65]:
```suggestion
            .map(|transaction| match transaction {
```

Well, `txn` will always mean DB transaction for me. But surely, not a blocker ;)

File: node/src/components/transaction_acceptor.rs [hunk 2]
```diff
@@ -0,0 +1,901 @@
...
+        block_header: Box<BlockHeader>,
+    ) -> Effects<Event> {
+        enum NextStep {
+            GetContract(ContractHash),
+            GetPackage(ContractPackageHash, Option<ContractVersion>),
+            CryptoValidation,
+        }
+
+        let next_step = match &event_metadata.transaction {
+            Transaction::Deploy(_) => unreachable!("should only handle transaction v1"),
```
Comment (rafal-ch) [inline node/src/components/transaction_acceptor.rs]: There are several `unreachable!`s in this PR. I see that they are in theory indeed unreachable, but I don't feel comfortable with these.

Anyway, definitely not a blocker, but please see if some of these could be dodged by some smart `match` or `if let...`.

File: node/src/components/storage.rs [hunk 3]
```diff
@@ -58,30 +58,35 @@ use datasize::DataSize;
 use derive_more::From;
 use itertools::Itertools;
 use lmdb::{
-    Cursor, Database, DatabaseFlags, Environment, EnvironmentFlags, RwTransaction, Transaction,
-    WriteFlags,
+    Cursor, Database, DatabaseFlags, Environment, EnvironmentFlags, RwTransaction,
+    Transaction as LmdbTransaction, WriteFlags,
```
Comment (rafal-ch) [inline node/src/components/storage.rs:62]: Ouch :)

File: node/src/components/storage.rs [hunk 4]
```diff
@@ -194,7 +202,10 @@ pub struct Storage {
     /// The state storage database.
     #[data_size(skip)]
     state_store_db: Database,
-    /// The finalized approvals database.
+    /// The finalized deploy approvals database.
+    #[data_size(skip)]
+    finalized_deploy_approvals_db: Database,
+    /// The finalized transaction approvals database.
     #[data_size(skip)]
     finalized_approvals_db: Database,
```
Comment (rafal-ch) [inline node/src/components/storage.rs]:
```suggestion
    finalized_transaction_approvals_db: Database,
```

File: node/src/effect.rs [hunk 5]
```diff
@@ -1890,78 +1900,38 @@ impl<REv> EffectBuilder<REv> {
...
+            },
+            QueueKind::ContractRuntime,
+        )
+        .await
     }
 
-    /// Retrieves an `ContractPackage` from global state if present.
-    pub(crate) async fn get_contract_package_for_validation(
+    /// Retrieves a `Package` from under the given key in global state if present.
+    pub(crate) async fn get_contract_package(
```
Comment (rafal-ch) [inline node/src/effect.rs]:
```suggestion
    pub(crate) async fn get_package(
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: 

Just some nits, but good to go as is. Nice job!

File: node/src/effect.rs (hunk 5 above)
Comment (EdHastingsCasperAssociation) [reply node/src/effect.rs]: @darthsiroftardis is supposed to be fixing up all the account / contract related renames in a later PR; i wouldn't bother nitpicking other ppl's PR's on the old names until after karan has done an across the board fixup.

File: node/src/components/fetcher/fetcher_impls/deploy_fetcher.rs (hunk 1 above)
Comment (Fraser999) [reply node/src/components/fetcher/fetcher_impls/deploy_fetcher.rs:65]: I have used `txn` pretty much everywhere except `Storage` and I think others will naturally use that abbreviation (or `tx` which I dislike as we generally use that for "transmit" in the context of channels), so I'd rather leave as is.

File: node/src/components/transaction_acceptor.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/components/transaction_acceptor.rs]: Resolved these in [66ceea6](https://github.com/casper-network/casper-node/pull/4236/commits/66ceea68545c848ce0bcfa7a2fbda6aa1dd769c4).

File: node/src/components/storage.rs (hunk 4 above)
Comment (Fraser999) [reply node/src/components/storage.rs]: Fixed in [66ceea6](https://github.com/casper-network/casper-node/pull/4236/commits/66ceea68545c848ce0bcfa7a2fbda6aa1dd769c4).

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Review by mpapierski — APPROVED** [review]
Looks good especially 3f6ff0e87df0e3235755754cb20cc06ecb5e8111

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**